src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -357,13 +357,23 @@ public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
 
                 }
 
+                // Check for extension options for the ROPC flow
+                TokenAcquisitionExtensionOptions? addInOptions = tokenAcquisitionExtensionOptionsMonitor?.CurrentValue;
+
                 // ROPC flow
-                var authenticationResult = await ((IByUsernameAndPassword)application).AcquireTokenByUsernamePassword(
-                    scopes.Except(_scopesRequestedByMsal),
-                    username,
-                    password)
-                    .ExecuteAsync()
-                    .ConfigureAwait(false);
+                AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder builder = ((IByUsernameAndPassword)application)
+                    .AcquireTokenByUsernamePassword(
+                        scopes.Except(_scopesRequestedByMsal),
+                        username,
+                        password);
```

Comment:
I think you want to relax the condition at line 334 to allow this code to be hit if the end-user provides an empty password for user FIC

Reply:
I was thinking of sending a default value for password from MISE when a user assertion is passed instead since MSAL also validates that the password should not be null.